pdns/backends/gsql/gsqlbackend.cc

```diff
@@ -341,23 +341,50 @@ void GSQLBackend::getUnfreshSlaveInfos(vector<DomainInfo> *unfreshDomains)
 
   for(const auto& row : d_result) { // id,name,master,last_check
     DomainInfo sd;
-    ASSERT_ROW_COLUMNS("info-all-slaves-query", row, 4);
+    try {
+      ASSERT_ROW_COLUMNS("info-all-slaves-query", row, 4);
```

Comment:
Why is this assert changed in a non fatal, very verbose one?

Reply:
before, this would lead to the whole `getUnfreshSlaveInfos` failing. _in theory_ other could be good still.

Reply:
In my opinion this is not the right place to fix this. Catch the exception and add some logging in getUnfreshSlaveInfos() is a better approach. When this assert is triggered it is likely for all rows in the result.
With a million domains this will result in a million lines of logging for every slave freshness check.

Reply:
I'll make sure it logs only once but keep the current semantics of "keep trying all other domains regardless"